src/lib/stitching/gravity/stitching.ts

```diff
@@ -288,6 +288,17 @@ export const gravityStitchingEnvironment = (
               ids = [null]
             }
 
+            // We can't send more than 100 artwork ids to /api/v1/artworks
+            let limit = 100
+            // Allow override of "first" and "last", but not if it's greater than 100
+            const arg_limit = args["first"] || args["last"]
+            if (arg_limit < limit) {
+              limit = arg_limit
+            }
+            if (limit < ids.length) {
+              ids = ids.slice(0, limit)
```

Comment:
If I understand correctly, this would only consider the first 100 artworks in a viewing room and ignore the rest. For example, if a viewing room has 101 artworks and client asks for the last 1, we'd return the 100th artwork instead of the 101st artwork.

If we are saying we only support fetching up to 100 artworks for a viewing room, maybe we should just limit the max number of artworks partners can add to a viewing room to be 100 (which seems reasonable to me, based on the Viewing Room experience)?

Reply:
This stitching (`artworksConnection`) is starting to be a little....nuanced 😅 .

For one, there's no clean way to handle a connection of > 100 artworks. Any truncation of id's here might not result in the expected page/sizes from the server, since some of these artworks might be unpublished or deleted.

Another, is that looking at the `artworks` resolver (https://github.com/artsy/metaphysics/blob/44b47d73e8919d299e50d1107a1511c7bdfc48f2/src/schema/v2/artworks.ts#L31), _all_ artworks are returned in the viewing room, and then only the `first: 20` (for instance) are returned to the client. But the Metaphysics <-> Gravity call loads all of them.

All this is to say that this type of stitching doesn't feel great. What if an old-school data loader and REST endpoint were used (for instance, `/api/v1/artworks?viewing_room_id=...`), and then this stitched resolver here would create a connection from that properly paginated call, etc. (basically the way we do it everywhere).

Essentially I _think_ this isn't an ideal use case of stitching, and feel like it would be better served with the more traditional loader/REST endpoint.